src/providers/cryptoauthlib/mod.rs

```diff
@@ -31,22 +31,33 @@ mod key_management;
 mod key_slot;
 mod key_slot_storage;
 
+// The UUID for this provider
+const PROVIDER_UUID: &str = "b8ba81e2-e9f7-4bdd-b096-a29d0019960c";
+
 /// CryptoAuthLib provider structure
 #[derive(Derivative)]
 #[derivative(Debug)]
 pub struct Provider {
     #[derivative(Debug = "ignore")]
     device: rust_cryptoauthlib::AteccDevice,
     provider_id: ProviderId,
+    // The name of the provider set in the config.
+    provider_name: String,
+    // The UUID of the provider.
+    provider_uuid: String,
```

Comment:
I just noticed: why is the provider UUID needed as a field here? Presumably that UUID is entirely static, so we could just use the `const` defined above.

Reply:
I added it so that we can use `self.provider_uuid` when declaring `KeyIdentity`s in the submodules.
Don't know if that makes any sense though or if there is a better way. I did think it needed moving out from just being a `&str` in the `describe` function though as we'll need to use it in multiple places.

Reply:
If you put it inside `impl Provider` (like the default name), you might be able to do `Self::PROVIDER_UUID` in submodules?

Reply:
Hmm, I think instead of doing that we could store the `const` value as a `Uuid` directly, using [one of the many `const` methods](https://docs.rs/uuid/0.8.2/uuid/struct.Uuid.html) implemented on Uuid. We can't use the current `parse_str`, but I think that's ok, we just need to make sure the current values are preserved. Then you can simply use that constant for `KeyIdentity`. Truth be told, I'd be inclined to not even have to pass that UUID as a parameter and simply populate it from the `KeyInfoManagerClient` like we do with the provider ID now.

Reply:
a `const Uuid` in the `impl` sounds like the best approach.